src/extension/gogoduck/src/main/java/au/org/emii/gogoduck/worker/SubsetParameters.java

```diff
@@ -0,0 +1,47 @@
+package au.org.emii.gogoduck.worker;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import au.org.emii.gogoduck.worker.SubsetParameters.SubsetParameter;
+
+public class SubsetParameters extends HashMap<String, SubsetParameter> {
```

Comment:
`getNcksParameters` needs to be thread safe. However it's not modifying the map, just iterating on it. I think it should be OK. No such thing as `const` functions in java, right?